### DISCUSSION TITLE
Set kernels for each distribution seperately

### DISCUSSION THREAD

**Discussion by zhengliuX** [microsoft/WSL#10238] [Q&A] [answered]
Hi, does WSL2 now support setting kernels for each distribution seperately? I have multiple distributions, like debian and centos, and I need to set kernel for each distribution.

**Comment by Biswa96** [comment, marked as answer]
> does WSL2 now support setting kernels for each distribution separately? 

No, it is not possible yet. Because the Linux kernel path is retrieved from .wslconfig file before starting the WSL2 VM. And there is only one WSL2 VM running.

But you can achieve that in different way, like start Debian > shutdown Debian > change Linux kernel path in .wslconfig > start CentOS.

**Reply by zhengliuX** [reply]
Thanks a lot for the reply! Is the feature in the development plan?

**Reply by Biswa96** [reply]
> Is the feature in the development plan?

Unfortunately, I am unware of that. Microsoft developers may provide more information.
